### PR TITLE
fix(overlays): only focus controlled elements after hide

### PR DESCRIPTION
**Pull request by tlouisse**
closes https://github.com/ing-bank/lion/issues/759

### CODE REVIEW

File: .changeset/lemon-shoes-doubt.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@lion/checkbox-group': patch
+---
+
+recover possibility to registrer without []
```
Comment (daKmoR) [inline .changeset/lemon-shoes-doubt.md:5]: huh why? 
this enforces a best practice... we no longer want to enforce it? 🤔

Comment (tlouisse) [reply .changeset/lemon-shoes-doubt.md:5]: @daKmoR this apparently already didn't work. When doing the types I reenabled it, but it breaks our extension layer (so a breaking change and reverting again).

Comment (tlouisse) [reply .changeset/lemon-shoes-doubt.md:5]: Somehow it never hit this line before: https://github.com/ing-bank/lion/blob/master/packages/checkbox-group/src/LionCheckboxGroup.js#L16
I actually think it also conflicts with https://github.com/ing-bank/lion/blob/master/packages/form-core/src/registration/FormRegistrarMixin.js#L155

So we probably should rediscuss before Lion v1

Comment (daKmoR) [reply .changeset/lemon-shoes-doubt.md:5]: ok, makes sense

**Review by daKmoR — APPROVED** [review]

**Pull request merged**